instrumentation/opentelemetry-instrumentation-annotations-1.16/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/instrumentationannotations/AnnotationInstrumentationModule.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.instrumentationannotations;
+
+import static java.util.Arrays.asList;
+
+import application.io.opentelemetry.instrumentation.annotations.WithCurrentSpan;
+import application.io.opentelemetry.instrumentation.annotations.WithSpan;
+import com.google.auto.service.AutoService;
+import io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule;
+import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
+import java.util.List;
+
+/**
+ * Instrumentation for methods annotated with {@link WithSpan} and {@link WithCurrentSpan}
+ * annotations.
+ */
+@AutoService(InstrumentationModule.class)
+public class AnnotationInstrumentationModule extends InstrumentationModule {
+
+  public AnnotationInstrumentationModule() {
+    super("opentelemetry-instrumentation-annotations");
+  }
+
+  @Override
+  public int order() {
+    // Run first to ensure other automatic intstrumentation is added after and therefore is executed
+    // earlier in the instrumented method and create the span to attach attributes to.
+    return Integer.MIN_VALUE;
```

Comment:
How about using -1000 or -10000 just in case someone wants to do something even before that in their custom extension/distro?
```suggestion
    return -1000;
```